### PR TITLE
Use registry for `v1.4.0` compatibility in `tce` prod harbor

### LINKED ISSUE #2535: Use production harbor for TKG Bom bits as defined in compatibility file
## Bug Report

We need to use the production harbor for all future releases of TCE:

https://github.com/vmware-tanzu/community-edition/blob/main/Makefile#L57

this was a fast workaround to get us upgraded to the latest `tanzu-framework` version which does _not_ have the latest compatible BOM artifacts from `tanzu-framework` release. For now, we can use the staging registry.

## Expected Behavior

Use production harbor during build and release

## Steps to Reproduce the Bug

When targeting the production harbor in our build:
```
$ CLUSTER_PLAN=dev tanzu management-cluster create mytest -i docker -v 10

Downloading TKG compatibility file from 'projects.registry.vmware.com/tkg/framework-zshippable/tkg-compatibility'
Error: unable to ensure prerequisites: unable to ensure tkg BOM file: failed to download default bom files from the registry: unable to get the default BOM file ImagePath and Image Tag from the TKG Compatibility file: unable to find the supported TKG BOM version for the management plugin version "v0.10.0" in the TKG Compatibility file "/home/ubuntu/.config/tanzu/tkg/compatibility/tkg-compatibility.yaml"
```

This is because the bom files are not present for the compatible version of `tanzu-framework` in the production harbor:
```
❯ cat ~/.config/tanzu/tkg/compatibility/tkg-compatibility.yaml
version: v20
managementClusterPluginVersions:
...
- version: v0.10.0
  supportedTKGBomVersions:
  - imagePath: tkg-bom
    tag: v1.5.0-tf-v0.10.0
...

❯ docker pull projects.registry.vmware.com/tkg/tkg-bom:v1.5.0-tf-v0.10.0
Error response from daemon: unknown: artifact tkg/tkg-bom:v1.5.0-tf-v0.10.0 not found
```

## Screenshots or additional information and context

Related to https://github.com/vmware-tanzu/community-edition/pull/2523

## Environment Details

N/a

## Diagnostics and log bundle
N/a

### PR DESCRIPTION
**Pull request by jpmcb**
## What this PR does / why we need it
This PR uses a compatibility file and bom from our `tce/` production harbor registry found at 
```
projects.registry.vmware.com/tce/compatibility/v0.10.0/tkg-compatibility
```

also includes several changes to capture using the compatibility file in the managed and standalone plugins.

## Details for the Release Notes (PLEASE PROVIDE)
```release-note
use compatibility file and bom for v1.4.0
```

## Which issue(s) this PR fixes
Fixes: https://github.com/vmware-tanzu/community-edition/issues/2535

## Describe testing done for PR
`tanzu management-cluster` creates cluster using the v1.4.0 bom

```
❯ CLUSTER_PLAN=dev tanzu management-cluster create mytest -i docker -v 10
Downloading TKG compatibility file from 'projects.registry.vmware.com/tce/compatability/v0.10.0-1/tkg-compatibility'
Downloading the TKG Bill of Materials (BOM) file from 'projects.registry.vmware.com/tce/tkg-bom:v1.4.0'
Downloading the TKr Bill of Materials (BOM) file from 'projects.registry.vmware.com/tkg/tkr-bom:v1.21.2_vmware.1-tkg.1'
```

`tanzu standalone-cluster` creates cluster using v1.4.0 bom

```
❯ ./artifacts/standalone-cluster/v0.10.0/tanzu-standalone-cluster-linux_amd64 create hello-world -i docker
...

Downloading TKG compatibility file from 'projects.registry.vmware.com/tce/compatibility/v0.10.0/tkg-compatibility'
Downloading the TKG Bill of Materials (BOM) file from 'projects.registry.vmware.com/tce/tkg-bom:v1.4.0'
Downloading the TKr Bill of Materials (BOM) file from 'projects.registry.vmware.com/tkg/tkr-bom:v1.21.2_vmware.1-tkg.1'
```

## Special notes for your reviewer
n/a

### CODE REVIEW

File: hack/builder/Makefile [hunk 1]
```diff
@@ -8,8 +8,12 @@ BUILD_SHA ?= $$(git rev-parse --short HEAD)
 BUILD_EDITION=tce-standalone
```
Comment (joshrosso) [inline hack/builder/Makefile:8]: could we leave builder unchanged? effectively shipping standalone exactly the way we did last release?

Comment (jpmcb) [reply hack/builder/Makefile:8]: We would need to roll back a big chunk of this PR then: https://github.com/vmware-tanzu/community-edition/commit/dccd271d08c092c69415d8e2dbcfa8cafe017c69#diff-178629e8a6889c3f38c18d928d26c514ca5486f801b8129b5821dfe3b6794b11

TLDR: we wouldn't be able to use the new LD Flag `buildinfo.Version` present in `tanzu-framework` v0.10.0 and would have to use `cli.BuildVersion` from v0.2.1. So there _is_ an unfortunate coupling between versions pulled in by the plugins and what we build with due to this LD flag.

Comment (jpmcb) [reply hack/builder/Makefile:8]: giving this a try now - rolling the builder back and will add as a new commit

Comment (jpmcb) [reply hack/builder/Makefile:8]: that didn't work. Rolled everything back to the older version of builder we use and still getting:

```
❯ tanzu standalone-cluster create hello-world -i docker

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
Warning - Standalone clusters will be deprecated in a future release of Tanzu Community Edition
                                   Use at your own Risk
           Checkout the proposal for the standalone cluster replacement:
           https://github.com/vmware-tanzu/community-edition/issues/2266
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Downloading TKG compatibility file from 'projects.registry.vmware.com/tce/compatability/v0.10.0/tkg-compatibility'
Error: unable to create Tanzu Standalone Cluster client
Cause: unable to ensure prerequisites: unable to ensure tkg BOM file: failed to download default bom files from the registry: unable to get the default BOM file ImagePath and Image Tag from the TKG Compatibility file: unable to find the supported TKG BOM version for the management plugin version "v0.2.1" in the TKG Compatibility file "/home/jmcb/.config/tanzu/tkg/compatibility/tkg-compatibility.yaml"
Error: exit status 1

✖  exit status 1

```

since we are passing the `TKG_DEFAULT_IMAGE_REPOSITORY` into the builder

File: Makefile [hunk 2]
```diff
@@ -54,7 +54,11 @@ TANZU_FRAMEWORK_REPO_BRANCH ?= v0.10.0
 TANZU_FRAMEWORK_REPO_HASH ?=
 # TKG_DEFAULT_IMAGE_REPOSITORY override for using a different image repo
 ifndef TKG_DEFAULT_IMAGE_REPOSITORY
-TKG_DEFAULT_IMAGE_REPOSITORY ?= projects-stg.registry.vmware.com/tkg
+TKG_DEFAULT_IMAGE_REPOSITORY ?= projects.registry.vmware.com/tce
+endif
+# TKG_DEFAULT_COMPATIBILITY_IMAGE_PATH override for using a different image path
+ifndef TKG_DEFAULT_COMPATIBILITY_IMAGE_PATH
+TKG_DEFAULT_COMPATIBILITY_IMAGE_PATH ?= compatability/v0.10.0-1/tkg-compatibility
```
Comment (jpmcb) [inline Makefile:61]: this is being thrown in the mispell checker:
```
Makefile:61:40: "compatability" is a misspelling of "compatibility"
```
do we care? or do we wanna push a new image?

Comment (jpmcb) [reply Makefile:61]: i'm pushing a new image now - please hold 👀

Comment (jpmcb) [reply Makefile]: Done! Pushed a new one at
```
compatibility/v0.10.0/tkg-compatibility
```

**Review by davidvonthenen — APPROVED** [review]
LGTM
